xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/exporter/exporter.js

```diff
@@ -152,60 +154,188 @@ require([
             },
             _disabled: !node.state.opened
           };
-          
+
           return items;
         };
 
         /**
          * Handle 'select all' button
          */
-        $('#exportModal .tree_select_all').click(function() {
+        $('#exportModal .tree_select_all').click(function () {
           treeReference.check_all();
           return false;
         });
-        
+
         /**
          * Handle 'select none' button
          */
-        $('#exportModal .tree_select_none').click(function() {
+        $('#exportModal .tree_select_none').click(function () {
           treeReference.uncheck_all();
           return false;
         });
-          
+
         /**
-         * Update the URL of the export buttons according to the tree 
+         * This function is called recursively to process each parent node in order to built properly the list of list
+         * of pages to includes (i.e. pages parameter of export) and pages to excludes (i.e. excludes parameter of
+         * export).
+         * FTR the parameters are ordered as the first element of pages is linked to the first element of excludes.
+         *
+         * @param rootNode: the parent node to process
+         * @param arrayOfPages: a list of list of pages to includes
+         * @param arrayOfExcludes: a list of list of pages to excludes relatively to their included pages
          */
-         $('#exportModal #exportModelOtherCollapse a.btn').click(function(event) {
-          var button = $(this);
-          var url = button.data('url').clone();
-          var checkedNodes = treeReference.get_checked(true);
-          if (url.params === undefined) {
-            url.params = {'pages': []};
+        var processLevel = function(rootNode, arrayOfPages, arrayOfExcludes) {
```

Comment:
If I understood correctly, the items from ``arrayOfExcludes`` are mapped to the items from ``arrayOfPages`` so it would have been nice to use a map data type to make this more explicit. Unfortunately [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) is not supported on older version of Internet Explorer and the plain JavaScript object has some reserved keys that theoretically could collide with the page references. In practice the page reference should contain at least one dot so there should be no collision. The only issues could be the page order, but I don't think it matters (Map preserves the key insertion order but we can't use it yet..)

Reply:
OK I'll refactor to use a POJO then.